.github/workflows/build.yml

```diff
@@ -5,18 +5,70 @@ on:
       - master
 
 jobs:
-  build:
+  docker-build:
+    name: Build & pubish Docker image
     runs-on: ubuntu-latest
     steps:
     - name: Checkout repo
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       with:
         ref: ${{ github.head_ref }}
 
-    - name: Docker
-      uses: docker/build-push-action@v1
+    # BUILDKIT_INLINE_CACHE creates the image in such a way that you can
+    # then use --cache-from (think of a remote cache)
+    # This feature can be used because of using the buildx plugin
+    # In the case of the Docker builder iamge, we should always see that it's using the remote cache
+    # NOTE: Sadly, the COPY line from the builder image (in the Dockerfile) always invalidates the cache,
+    # thus, we do not benefit from using --cache-from. The multi-stage build still gives us a smaller footprint
+    # TODO: Optimize the Dockerfile to take advantage of proper caching
+    - name: Docker build images
+      run: |
+        docker buildx build . \
+          --build-arg BUILDKIT_INLINE_CACHE=1 \
+          -t ghcr.io/getsentry/action-release-image:latest \
+          --cache-from ghcr.io/getsentry/action-release-image:latest \
+          --target app
+
+    # Test Docker build before publishing
+    - name: Use local Dockerfile for action
+      uses: './.github/actions/use-local-dockerfile'
+
+    - name: Mock creating a Sentry release
+      uses: ./
+      env:
+        MOCK: true
       with:
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
-        repository: sentryintegrations/sentry-github-action-release
-        tags: latest
+        environment: production
+
+    - name: Login and publish
+      run: |
+        echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
```

Comment:
neat little shell trick -- `docker login ... <<< ${{ secrets.GITHUB_TOKEN }}` -- https://youtu.be/Xf_82stIbB8